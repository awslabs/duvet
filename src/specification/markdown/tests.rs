// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

use super::*;

fn tokens(contents: &str) -> Vec<Token> {
    Lex::new(contents).collect()
}

macro_rules! snapshot {
    ($name:ident, $contents:expr) => {
        mod $name {
            #[test]
            fn tokens() {
                insta::assert_debug_snapshot!("tokens", super::tokens($contents));
            }

            #[test]
            fn tree() {
                insta::assert_debug_snapshot!("tree", super::parse($contents));
            }
        }
    };
}

snapshot!(
    simple,
    r#"
# This is a test

Content goes here. Another
sentence here.
"#
);

snapshot!(
    multi_line_header,
    r#"
Foo *bar
baz*
======

Content goes here. Another
sentence here.
"#
);

snapshot!(
    multiple,
    r#"
# This is a test

Content goes here. Another
sentence here.

## This is another test

More content goes here

### Nested section

Testing 123

## Up one

Another section
"#
);

snapshot!(
    list_example,
    r#"
# List example

Here is a list:
* Item 1
* Item 2
  * Item 2.1
* Item 3
  * Item 3.1
    * Item 3.1.1
    * Item 3.1.2
  * Item 3.2

Here is a numbered list:
1. Item 1
2. Item 2
3. Item 3

Here is a list with content:
* Item
  More content

  Other content

  * Testing

    Other test

Testing 123

* Item
More content
"#
);
