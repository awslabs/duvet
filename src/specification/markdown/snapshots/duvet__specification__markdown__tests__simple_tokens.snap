---
source: src/specification/markdown/tests.rs
assertion_line: 20
expression: "tokens(r#\"\n# This is a test\n\nContent goes here\n\n## This is another test\n\nMore content goes here\n\n### Nested section\n\nTesting 123\n\n## Up one\n\nAnother section\n\"#)"
---
[
    Line(
        Str {
            value: "",
            pos: 0,
            line: 1,
        },
    ),
    Header(
        Str {
            value: "# This is a test",
            pos: 1,
            line: 2,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 18,
            line: 3,
        },
    ),
    Line(
        Str {
            value: "Content goes here",
            pos: 19,
            line: 4,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 37,
            line: 5,
        },
    ),
    Header(
        Str {
            value: "## This is another test",
            pos: 38,
            line: 6,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 62,
            line: 7,
        },
    ),
    Line(
        Str {
            value: "More content goes here",
            pos: 63,
            line: 8,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 86,
            line: 9,
        },
    ),
    Header(
        Str {
            value: "### Nested section",
            pos: 87,
            line: 10,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 106,
            line: 11,
        },
    ),
    Line(
        Str {
            value: "Testing 123",
            pos: 107,
            line: 12,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 119,
            line: 13,
        },
    ),
    Header(
        Str {
            value: "## Up one",
            pos: 120,
            line: 14,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 130,
            line: 15,
        },
    ),
    Line(
        Str {
            value: "Another section",
            pos: 131,
            line: 16,
        },
    ),
]
